### ISSUE TITLE
Build of Ipopt fails on Debian Linux

### ISSUE THREAD

**Issue by leethargo** [JuliaOpt/Ipopt.jl#66]
Possibly a duplicate of #62, but I don't have the `lib64` folders.

It fails during the build of dependency Lapack:
```
INFO: Changing Directory to /home/rs/.julia/v0.4/Ipopt/deps/src/Ipopt-3.12.1/ThirdParty/Lapack/build
[...]
checking for pkg-config... no
checking whether -lblas has BLAS... no
checking for COIN-OR package Blas... skipped check via pkg-config, redirect to fallback
checking for COIN-OR package Blas (fallback)... no, dependency coinblas not available
configure: error: Required package BLAS not found.
```

But BLAS was built just before, and it seems successfully:
```
julia> Pkg.dir("Ipopt")
"/home/rs/.julia/v0.4/Ipopt"

shell> ls /home/rs/.julia/v0.4/Ipopt/
deps  doc  example  LICENSE.md  README.md  REQUIRE  src  test

shell> ls /home/rs/.julia/v0.4/Ipopt/deps
build.jl  downloads  src  usr

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr
lib

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/
libcoinblas.a  libcoinblas.la  pkgconfig

shell> ls /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/
coinblas.pc

shell> cat /home/rs/.julia/v0.4/Ipopt/deps/usr/lib/pkgconfig/coinblas.pc
prefix=/home/rs/.julia/v0.4/Ipopt/deps/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib

Name: Blas
Description: Basic linear algebra package
URL: https://projects.coin-or.org/BuildTools
Version: 1.4.2
Libs: -L${libdir} -lcoinblas  -L/usr/lib/gcc/x86_64-linux-gnu/4.9 -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/4.9/../../.. -lgfortran -lm -lquadmath
Cflags:
```

**Comment by mlubin** [comment]
If this is a recent version of debian you could try using the builtin ipopt
package instead.

**Comment by leethargo** [comment]
Thanks, that worked for me. :+1:

**Comment by tkelman** [comment]
would still like to see the config.log to see what happened

**Comment by leethargo** [comment]
I uploaded the [config.log](https://gist.github.com/leethargo/822e62b67db22f1f75b77dc3698fa072). I tried both the `Ipopt.jl` release and then the `master`, with the same behavior.

**Comment by tkelman** [comment]
try adding `--with-blas="$prefix/lib/libcoinblas.a -lgfortran"` to the lapack configure line?

though nowadays since we're renaming symbols in openblas we might not need to do this static blas thing. maybe still useful if you want to use ilp64 mkl with julia since ipopt would need lp64 and mkl uses the same symbol names.

**Comment by leethargo** [comment]
@tkelman: With these extra params to the configure it works for me, too.

**Comment by tkelman** [comment]
And you're sure it's using the source build this time? `deps.jl` contains that path?

**Comment by leethargo** [comment]
I uninstalled the Debian package and called `Pkg.build("Ipopt")`. A lot of compilation followed, I can paste the log.

**Comment by tkelman** [comment]
Great, thanks for testing that! I'll file that change as a PR then.

**Comment by tkelman** [comment]
installing pkg-config might have also been a way to fix this, but #67 should make it unnecessary
